CHANGELOG.md

```diff
@@ -2,11 +2,197 @@
+    - The sniff presumed the WP naming conventions for PHP constructs, while it should check for the construct being prefixed regardless of whether camelCase, PascalCase, snake_case or other naming conventions are used.
+    - The sniff presumed the WP naming conventions for prefixes used in hook names. The sniff will now be more tolerant when non-conventional word separators are used in prefixes for hooks.
+- The `WordPress.NamingConventions.ValidFunctionName` sniff no longer "hides" one message behind another. The sniff will now correctly throw a message about function names not being in `snake_case`, even when the `FunctionDoubleUnderscore` or `MethodDoubleUnderscore` errorcodes have been excluded.
+- The `WordPress.PHP.StrictInArray` sniff will no longer throw an error when `in_array`, `array_search` or `array_keys` are used in a file `use` statement.
+- Various bugs in the `WordPress.Security.EscapeOutput` sniff:
+    - A limited list of native PHP constants which are safe to use, such as `PHP_EOL`, has been added. When any of these constants are encountered, the sniff will no longer demand output escaping for them.
+    - The sniff was underreporting issues with variables passed to `trigger_error()`.
+    - While reporting an issue, sometimes the wrong error message was used. The sniff logic has been adjusted to prevent this.
+    - The sniff will now correctly ignore the open and close brackets of short arrays.
+    - The sniff would throw false positives when `echo`, `print`, `exit` or `die` were encountered as constants, function or class names. While it may not be a good idea to use PHP keywords in such a way, it is allowed, so the sniff should handle this correctly.
+- The `WordPress.WhiteSpace.ControlStructureSpacing` sniff would inadvertly throw an error for the spacing around the colon for a return type in a function declaration.
+- The `WordPress.WP.AlternativeFunctions` sniff used to flag all function calls to `file_get_contents()` twice, suggesting to use `wp_remote_get()` - which is only applicable for remote URLs - and the `WP_FileSystem` API - which is not needed when just _reading_ local files. These messages contradicted each other.
+    The sniff will now try to determine whether the file requested is local or remote and will only throw a `warning` suggesting to use `wp_remote_get()`, if a remote URL is being requested or when it could not be determined if the requested file is local or remote.
+- The expected default value for `wp_upload_bits()` in the `WordPress.WP.DeprecatedParameters` sniff.
+- The `WordPress.WP.GlobalVariablesOverride` sniff previously did not detect variables created by a `foreach()` construct which would override WP global variables.
+- Various bugs in the `WordPress.WP.I18n` sniff:
+    - The sniff will no longer throw false positives for calls to methods carrying the same name as any of the global WP functions being targetted and has improved handling of parse errors and live coding.
```

Comment:
To form the past participle of the verb target, the final consonant should not be doubled before adding -⁠ed. targetted => targeted.